### ISSUE TITLE
build_best versus build_merged what is the difference if no testset is given.

### ISSUE THREAD

**Issue by JanoschMenke** [MolecularAI/QSARtuna#39]
I just wanted to ask what the difference between ` build_best` versus `build_merged` is, when there is no testset specified such as here

```
config = OptimizationConfig(
    data=Dataset(
        input_column="canonical",
        response_column="molwt",
        training_dataset_file="tests/data/DRD2/subset-50/train.csv",
        split_strategy= Random(),
    ),
)
```

Based on the results I get I assume that build_merged is trained on the complete `training_dataset` supplied, while build best is trained only on a subset, I assume generated by the `split_strategy` but when I am using a 10-fold criss validation which of the 10 fold is it split by?

```
# Build (re-Train) and save the best model.
build_best(buildconfig, "target/best.pkl")

# Build (Train) and save the model on the merged train+test data.
build_merged(buildconfig, "target/merged.pkl")
```

**Comment by lewismervin1** [comment]
Hi @JanoschMenke, I want to clarify what the `split_strategy` belonging to the Dataset class and the `cv_split_strategy` supplied by the Settings class to the `OptimizationConfig` do and how they behave as two different split implementations. 

1.) The `cv_split_strategy` is simply the strategy that is used by the hyper-parameter cross validation to obtain the performance of different parameter trials, which defaults to `Random`, e.g. as in your example.

2.) The `split_strategy` which is `Random()` in your example will split your train set into one train and one (outer) test set which is not used in the optimisation to avoid leakage. Any performance will be reported on the data reserved by this "Random" test set.

For your example, if you perform `build_best` you will train/build a model with the parameters from the highest performing hyper-parameter trial on only the train data (nothing in your random held out split).  Your model is evaluated on the test data.

In comparison, if you perform `build_merged` you will obtain a model with both the "Random" test/train split combined again (with the caveat that no performance evaluation will be reported).

You may (optionally) provide a `test_dataset_file` to the Dataset class. This may be an additional holdout test dataset to evaluate an optimized model outside of the Optuna Hyperparameter search, similar to 2.) above. It may be used if `split_strategy` is set to NoSplitting and may be left empty if splitting strategy is set to split the data. It is even possible to supply both split_strategy and test_dataset_file (which you may have been doing), and then the two test sets from your file and from your split_strategy are combined, and performance is evaluated (and weighted equally) on both test sets in a single metric calculation (i.e. they are not calculated and aggregated together). 

If you provide neither `split_strategy` nor `test_dataset_file`, then we observe no difference between build_best and build_merged.

I hope this clarifies? Please do let me know if not.
